src/jedi_bundle/bin/jedi_bundle.py

```diff
@@ -178,6 +188,13 @@ def jedi_bundle():
         # ---------------
         config_dict = load_yaml(logger, config_file_name)
 
+        # If pinned_versions exists, concatenate with config_dict if not already there
+        # ----------------------------------------------------------------------------
+        if pinned_versions_yaml:
+            if 'pinned_versions' not in config_dict:
```

Comment:
Can `pinned_versions` also be passed directly to the config file (`build.yaml`) as well?

If not, then this should probably throw an error; e.g.,

```python
assert "pinned_versions" not in config_dict
```

In general, I think `pinned_versions` should live _either_ in a dedicated file (in which case, we just throw the error above) or in an existing config file (in which case, I would remove the dedicated pinned versions file and just pull this information directly from the config dict), but probably not both (to avoid user confusion and having to build in and document inheritance).

But, if we want to keep the flexibility of both, we need to be clear and loud about inheritance/priority. E.g.,

```python
if pinned_versions_yaml:
    if 'pinned_versions' in config_dict:
        logger.info("Overwriting existing `pinned_versions` info in `config_dict` with YAML file")
    # Note: *Always* overwrite the existing dict with the YAML file, to avoid ambiguity
    pinned_versions_dict = load_yaml(logger, pinned_versions_yaml)
    config_dict['pinned_versions'] = pinned_versions_dict
```

If so, a few thoughts:

(1) Is there a good reason we want to have a _separate_ file specifically for `pinned_versions` vs. just keeping everything in the `build.yaml`? I would be inclined to keep as much of the configuration in a single config file as possible and avoid creating extra config files. But, if it's better to keep the pinned versions in a dedicated YAML file, I would be inclined to _remove_ 

(2)

Reply:
The only reason line 193-196 was added is to handle calling the clone step with the pinned versions argument (i.e. `jedi_bundle Clone --pinned_versions /path/to/pinned_versions.yaml build.yaml`). This is only necessary for the github action test. Otherwise, the pinned_version.yaml is appended to the build.yaml when you call `jedi_bundle --pinned_versions /path/to/pinned_versions.yaml`. 

What I can do is remove lines 193-196 and then devise a test that checks if pinned_versions was added to `build.yaml` in the github action.

I can make a dedicated pinned_versions file in jedi_bundle, that makes sense